java/vector/src/main/java/org/apache/arrow/vector/ZeroVector.java

```diff
@@ -109,7 +111,7 @@ public void clear() {
 
   @Override
   public Field getField() {
-    return new Field(name, FieldType.nullable(new Null()), null);
+    return new Field(DATA_VECTOR_NAME, FieldType.nullable(new Null()), null);
```

Comment:
I don't think we should fix the name here to `BaseRepeatedValueVector.DATA_VECTOR_NAME`.  The default should just be a generic name, I think what was there before was fine.

Reply:
This causes weird problems if you're trying to merge different schemas since it suggests that the inner vector changed if we moved from a List<null> to a List<int>, thus the change. The idea that the vector name is different before and after we populate the list doesn't really make any sense. Inner vectors should always have the same name.

Reply:
That's fine, but my concern was that is `ZeroVector` only meant to ever be an inner data vector?  Or would a user ever want to make a ZeroVector and not name it `DATA_VECTOR_NAME`? If not then I think we should just hardcode the name "$data$" like you mentioned before. Otherwise maybe we can do something like allow setting a name in the constructor, and create the instance with the name "$data$".